plugin/src/plugin.ts

```diff
@@ -6,32 +6,51 @@ import type { ExplicitWorklet, ReanimatedPluginPass } from './types';
 import { processIfWorkletNode } from './processIfWorkletNode';
 import { processInlineStylesWarning } from './processInlineStylesWarning';
 import { processIfCallback } from './processIfCallback';
+import { isRelease } from './utils';
 
 module.exports = function (): PluginItem {
+  function callGuardDEV(fun: () => void) {
+    try {
+      fun();
+    } catch (e) {
+      throw new Error('[Reanimated] Babel plugin exception: ' + e);
+    }
+  }
+
+  function releaseCaller(fun: () => void) {
+    fun();
+  }
+
+  const maybeUseCallGuardDEV = isRelease() ? releaseCaller : callGuardDEV;
+
   return {
     pre() {
-      // allows adding custom globals such as host-functions
-      if (this.opts != null && Array.isArray(this.opts.globals)) {
-        this.opts.globals.forEach((name: string) => {
-          globals.add(name);
-        });
-      }
+      maybeUseCallGuardDEV(() => {
+        // allows adding custom globals such as host-functions
+        if (this.opts != null && Array.isArray(this.opts.globals)) {
+          this.opts.globals.forEach((name: string) => {
+            globals.add(name);
+          });
+        }
```

Comment:
I know this is out of scope for this PR but could we move it to a separate function for the purpose of consistency?